pandas/core/groupby/grouper.py

```diff
@@ -496,11 +496,16 @@ def __init__(
             # In extant tests, the new self.grouping_vector matches
             #  `index.get_level_values(ilevel)` whenever
             #  mapper is None and isinstance(index, MultiIndex)
+            # TODO: Can you have two levels with the same name?
```

Comment:
Looks like you can

```
In [8]: pd.MultiIndex.from_tuples([(1, 2), (3, 4)], names=["f", "f"])
Out[8]:
MultiIndex([(1, 2),
            (3, 4)],
           names=['f', 'f'])

In [9]: pd.MultiIndex.from_tuples([(1, 2), (3, 4)], names=["f", "f"]).names
Out[9]: FrozenList(['f', 'f'])
```

Reply:
Thanks! I was curious how this code worked if there were duplicate labels. If a label is provided, it takes the first level that matches the label. I'm wondering if we should raise instead (as it's ambiguous), but this PR doesn't change that behavior.

Reply:
Would the analogous operation be grouping on a DataFrame with duplicate column labels?

```
In [1]: df = pd.DataFrame(np.eye(3), columns=list("abb"))

In [2]: df
Out[2]:
     a    b    b
0  1.0  0.0  0.0
1  0.0  1.0  0.0
2  0.0  0.0  1.0

In [3]: df.groupby("b").mean()
ValueError: Grouper for 'b' not 1-dimensional
```

Reply:
Good point - this is making me more convinced we should raise if the groupers are in the index. I think this would need a deprecation in 2.0.

Reply:
Opened #49434